virttest/libvirt_xml/vm_xml.py

```diff
@@ -563,17 +563,19 @@ def del_controller(self, controller_type=None):
         for controller in del_controllers:
             self.xmltreefile.remove(controller)
 
-    def get_controllers(self, controller_type=None):
+    def get_controllers(self, controller_type=None, model=None):
         """
-        Get controllers according controller type
+        Get controllers according controller type and/or model type
 
         :param controller_type: type of controllers need to get
+        :param model: model of controllers need to get
         :return: controller list
         """
         all_controllers = self.xmltreefile.findall("devices/controller")
         type_controllers = []
         for controller in all_controllers:
-            if controller.get("type") != controller_type:
+            if ((controller_type is not None and controller.get("type") != controller_type) or
```

Comment:
there is one hidden risk issue, previously if controller_type == none, it will go into if statement, however  after you change, it absolutely not enter if statement. Is it intended by you?

Reply:
Yes, it is intended by me. 
Before, when controller_type == none, there will be NO matching controllers, it is not reasonable.
controller_type == none means controller_type is not specified and thus any controller type is qualified.

Reply:
there may be some existing test case code ,where  call get_controllers() without giving any parameters, and previous it will not return any controller back, but after your change , it now return controller directly. It is against previous function purpose.

Reply:
Searched get_controllers() in master branch, it is not called anywhere by now.